SilKit/source/services/orchestration/TimeSyncService.cpp

```diff
@@ -546,6 +612,75 @@ bool TimeSyncService::AbortHopOnForCoordinatedParticipants() const
     }
     return false;
 }
+
+auto TimeSyncService::GetCurrentWallClockSyncPoint() const -> std::chrono::nanoseconds
+{
+    Lock lock{_mx};
+    return _currentWallClockSyncPoint;
+}
+
+void TimeSyncService::StartWallClockCouplingThread()
+{
+    _wallClockCouplingThreadRunning = true;
+    _stopFuture = _wallClockCouplingThreadStopPromise.get_future();
```

Comment:
It is probably better to not store the future as a member variable, but only on the stack and move it into the lambda.

```c++
auto stopFuture = _wallClockCouplingThreadStopPromise.get_future();

_wallClockCouplingThread = std::thread{[stopFuture = std::move(stopFuture)]() mutable {
    ...
}};
```